pkg/controller/v1beta1/trainedmodel/controller.go

```diff
@@ -64,22 +63,66 @@ func (r *TrainedModelReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		}
 		return reconcile.Result{}, err
 	}
-	log.Info("Reconciling TrainedModel", "apiVersion", tm.APIVersion, "trainedmodel", tm.Spec)
-	shardStrategy := memory.MemoryStrategy{}
-	shardId := shardStrategy.GetOrAssignShard(tm)
-	// Use tm's parent InferenceService field to get the model modelConfig
-	modelConfigName := constants.ModelConfigName(tm.Spec.InferenceService, shardId)
-	modelConfig := &v1.ConfigMap{}
-	if err := r.Get(context.TODO(), types.NamespacedName{Name: modelConfigName, Namespace: req.Namespace}, modelConfig); err != nil {
-		log.Error(err, "Failed to find model ConfigMap to reconcile for InferenceService", "name", tm.Spec.Model, "namespace", req.Namespace)
-		// Error reading the object - requeue the request.
+
+	// If the parent InferenceService does not exists, delete the trainedmodel
+	isvc := &v1beta1api.InferenceService{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Namespace: req.Namespace, Name: tm.Spec.InferenceService}, isvc); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Parent InferenceService does not exists, deleting TrainedModel", "TrainedModel", tm.Name, "InferenceService", isvc.Name)
+			r.Delete(context.TODO(), tm)
+			return reconcile.Result{}, nil
+		}
 		return reconcile.Result{}, err
 	}
+	// Set TrainedModel's ownerReference to its parent InferenceService so when the parent InferenceService is removed
+	// the TraineModel object will be removed
+	log.Info("add owner reference")
+	if err := controllerutil.SetControllerReference(isvc, tm, r.Scheme); err != nil {
```

Comment:
Did you check if the owner reference is set correctly, I believe you will need to do an update here for the isvc.

Reply:
The ownerreference is added but when I delete InferenceService the trainedmodel was not removed. Let me try adding an update.

Reply:
I didn't need to send an update for isvc when I set ownerReference for model configmap and the configmap is garbage collected correctly:
```
if err := controllerutil.SetControllerReference(isvc, newModelConfig, c.scheme); err != nil {
	return err
}
```